appserver/static/recordingbtn.js

```diff
@@ -5,22 +5,54 @@ require([
     'splunkjs/mvc/tableview',
     'splunkjs/mvc/simplexml/ready!'
 ], function(_, $, mvc, TableView) {
-    // Translations from rangemap results to CSS class
+
+    var submittedTokens = mvc.Components.getInstance('submitted');
+
+    function callIdDrilldown(callId){
+        console.log("callIdDrilldown");
+        let earliestTime = submittedTokens.get('timeRange.earliest');
+        let latestTime = submittedTokens.get('timeRange.latest');
+        let link = `search?q=%60default_index%60%20sourcetype%3D%223cx%3Acalls%22%20call_id%3D${callId}&earliest=${earliestTime}&latest=${latestTime}`;
+        window.open(link, '_blank');
+    }
+    window.callIdDrilldown = callIdDrilldown;
+    function recordingButtonClick(link){
+        console.log("recordingButtonClick");
+        window.open(link, '_blank');
+    }
+    window.recordingButtonClick = recordingButtonClick;
+
     var TBRenderer = TableView.BaseCellRenderer.extend({
         canRender: function(cell) {
             // Only use the cell renderer for the range field
-            return cell.field === 'Recording';
+            return cell.field === 'call_id' || cell.field === 'Recording';
         },
         render: function($td, cell) {
             // Create the button with the URL to put in the cell
-	    if(cell.value === 'NOREC') {
-               $td.html('No recording found');
-	    } else {
-               $td.html('<a class="btn" href='+cell.value+'>Play Recording</a>');
-	    }
+            if(cell.field === 'call_id'){
+                $td.html('<a target="_blank" target="_blank" onClick="window.callIdDrilldown(\''+cell.value+'\')">'+cell.value+'</a>');
+            }
+            else if(cell.field === 'Recording'){
+                if(cell.value === null) {
+                    $td.html('');
+                } else {
+                    if(cell.value instanceof Array){
+                        let inner_html = '';
+                        for(let i=1; i<cell.value.length; i++){
```

Comment:
value of **variable i** should be start with 0

Reply:
Nice catch bro, Done!!